tool/tsh/tsh.go

```diff
@@ -3289,6 +3290,16 @@ func makeClientForProxy(cf *CLIConf, proxy string, useProfileLogin bool) (*clien
 		cf.AuthConnector = constants.HeadlessConnector
 	}
 
+	if cf.AuthConnector == constants.HeadlessConnector {
+		// Lock the process memory to prevent rsa keys and certificates from being exposed in a swap.
+		if err := mlock.LockMemory(); err != nil {
+			if trace.IsNotImplemented(err) {
+				return nil, trace.BadParameter("headless login is not supported on non-linux operating systems")
```

Comment:
@Joerger Please skip this check on Windows and MacOS. We give this as additional security on Linux as mlock is supported there. I don't think that other OSes should be excluded from the list only because of that.